test.py

```diff
@@ -362,18 +368,56 @@ class BoostTestSuite(UnitTestSuite):
     # --list_content. Static to share across all modes.
     _case_cache: Dict[str, List[str]] = dict()
 
+    _exec_name_cache: Dict[str, str] = dict()
+
+    def _generate_cache(self) -> None:
+        # Apply combined test only for test/boost
+        if self.name != 'boost':
+            return
+        exe = path_to(self.mode, "test", self.name, 'combined_tests')
```

Comment:
This makes the other boost tests to fail if combined_tests is not built.

$ ninja build/debug/test/boost/big_decimal_test
$ ./test.py -s --repeat 1 --mode debug -v  boost/big_decimal_test

```
[5/6] LINK (stripped) build/debug/test/boost/big_decimal_test
Traceback (most recent call last):
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 2252, in workaround_python26789
    code = await main()
           ^^^^^^^^^^^^
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 1946, in main
    await find_tests(options)
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 1597, in find_tests
    suite = TestSuite.opt_create(f, options, mode)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 217, in opt_create
    suite = SpecificTestSuite(path, cfg, options, mode)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 401, in __init__
    self._generate_cache()
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 377, in _generate_cache
    exe = path_to(self.mode, "test", self.name, 'combined_tests')
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/asias/src/cloudius-systems/scylla/./test.py", line 100, in path_to
    raise FileNotFoundError(f"{exe_path} does not exist.")
FileNotFoundError: build/debug/test/boost/combined_tests does not exist.
```

Reply:
@syuu1228 FYI.